src/state/actions/dispatchArIOInteraction.ts

```diff
@@ -123,6 +126,92 @@ export default async function dispatchArIOInteraction({
           WRITE_OPTIONS,
         );
         break;
+      case ARNS_INTERACTION_TYPES.PRIMARY_NAME_REQUEST: {
+        dispatch({
+          type: 'setSigningMessage',
+          payload: 'Confirming Primary Name 1/2',
+        });
+        const existingPrimaryNameRequest = await arioContract
+          .getPrimaryNameRequest({
+            initiator: owner.toString(),
+          })
+          .catch((e) => {
+            console.error(e);
+            return undefined;
+          });
+
+        if (!existingPrimaryNameRequest) {
+          await arioContract
+            .requestPrimaryName({
+              name: payload.name,
+            })
+            .catch((e) => {
+              throw new Error('Unable to request Primary name: ' + e.message);
+            });
+        } else if (existingPrimaryNameRequest.initiator !== owner.toString()) {
+          throw new Error(
+            `Primary name is currently being requested by ${existingPrimaryNameRequest.initiator}`,
+          );
+        }
+
+        let storedNameRequest: AoPrimaryNameRequest | boolean | undefined =
+          undefined;
+        // do to latency we retry for some time here - may need to adjust time
+        const storeRequestTimeout = setTimeout(() => {
+          if (storedNameRequest === undefined) storedNameRequest = false;
+        }, 1000 * 60 * 3);
+
+        while (storedNameRequest === undefined) {
+          const primaryNameRequest = await arioContract
+            .getPrimaryNameRequest({
+              initiator: owner.toString(),
+            })
+            .catch((e) => {
+              console.error(e);
+              return undefined;
+            });
+          if (
+            primaryNameRequest?.name &&
+            primaryNameRequest.name !== payload.name
+          ) {
+            throw new Error(
+              'Another Primary Name request for ' +
+                primaryNameRequest.name +
+                ' is conflicting with this request.',
+            );
+          }
+          storedNameRequest = primaryNameRequest;
+        }
+        clearTimeout(storeRequestTimeout);
```

Comment:
not sure all this logic is necessary - you can create the request and approve it with sequential async calls. no need t check state in between. a successful response from the contract indicates the request was properly stored